libsweep/src/sweep.cc

```diff
@@ -288,111 +235,72 @@ void sweep_device_accumulate_scans(sweep_device_s device) {
       received = 1;
     }
   }
+} catch (const std::exception& e) {
+  // worker thread is dead at this point
```

Comment:
Ok, so what I meant was that errors in the worker thread (I.e. Inside `accumulate_scans`) could be stored in the queue (e.g. via an additional data member) and then be presented to the user at the next call to `get_scan`